### ISSUE TITLE
downloadFile promise does not reject nor resolve on no network connection

### ISSUE THREAD

**Issue by br4nnigan** [johanneslumpe/react-native-fs#239]
which is pretty bad?

**Comment by cjdell** [comment]
@br4nnigan Agreed, am in the process of tidying the networking site. Will add a timeout option or network detection of some sort.

**Comment by br4nnigan** [comment]
found another similar issue: 

try url "http://google.com/asd.txt", you'll get a 404
try url "http://google" and you'll get nothing again

**Comment by br4nnigan** [comment]
adding to this: on ANDROID you get a "no such file" exception on the target local path, for both cases in my previous post.

**Comment by br4nnigan** [comment]
same for requests that return http status 300 (304 too iirc)

**Comment by Macromeda** [comment]
any updates ?

**Comment by amrlsayed** [comment]
any updates ?
